File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/ssm/api/fabric/utils/InvokeArgsUtils.java
package io.komune.ssm.api.fabric.utils;

import io.komune.ssm.api.fabric.model.InvokeArgs;

public class InvokeArgsUtils {

    public static String BLOCK_QUERY = "block";
    public static String LIST_QUERY = "list";
    public static String TRANSACTION_QUERY = "transaction";
    public static String QUERY_FUNCTION = "query";

    public static Boolean isBlockQuery(InvokeArgs args) {
        return isQueryOfType(args, BLOCK_QUERY);
    }

    public static Boolean isListQuery(InvokeArgs args) {
        return args.getFunction().equalsIgnoreCase(LIST_QUERY);
    }

    public static Boolean isTransactionQuery(InvokeArgs args) {
        return isQueryOfType(args, TRANSACTION_QUERY);
    }

    private static Boolean isQueryOfType(InvokeArgs args, String type) {
        return !args.getValues().isEmpty() && (
            args.getFunction().equalsIgnoreCase(type) || args.getValues().get(0).equalsIgnoreCase(type)
        );
    }

    public static Boolean isQueryFunction(InvokeArgs args) {
        return args.getFunction().equalsIgnoreCase(QUERY_FUNCTION);
    }

}

File: c2-chaincode/chaincode-dsl/src/commonMain/kotlin/io/komune/c2/chaincode/dsl/invoke/InvokeArgsUtils.kt
package io.komune.c2.chaincode.dsl.invoke


object InvokeArgsUtils {
    var BLOCK_QUERY: String = "block"
    var LIST_QUERY: String = "list"
    var TRANSACTION_QUERY: String = "transaction"
    var QUERY_FUNCTION: String = "query"

    fun isBlockQuery(args: InvokeArgs): Boolean {
        return isQueryOfType(args, BLOCK_QUERY)
    }

    fun isListQuery(args: InvokeArgs): Boolean {
        return args.function.equals(LIST_QUERY, ignoreCase = true)
    }

    fun isTransactionQuery(args: InvokeArgs): Boolean {
        return isQueryOfType(args, TRANSACTION_QUERY)
    }

    private fun isQueryOfType(args: InvokeArgs, type: String): Boolean {
        return args.values.isNotEmpty() && (args.function.equals(type, ignoreCase = true) || args.values
            .get(0).equals(type, ignoreCase = true)
                )
    }

    fun isQueryFunction(args: InvokeArgs): Boolean {
        return args.function.equals(QUERY_FUNCTION, ignoreCase = true)
    }
}
